text/0000-interactive-app-creation.md

```diff
@@ -0,0 +1,142 @@
+
+1. `ember new`
+1. `ember new --interactive`
+1. `ember new -i`
+1. `ember new my-app -i`
+
+### App/Addon 
+
+Starting a new Ember app or addon as we do now (at the time of this writing) will not change. If a user types `ember new my-app` they will still get a new Ember app generated in the usual ways. However, if the user types `ember new` they will enter into an interactive workflow that will ask them a few key questions about their application. 
+
+If a user were to type in existing flags that are also questions in the interactive workflow (i.e., `ember new my-app --lang en-US --interactive`), it will skip the question entirely. 
+
+When a user types `ember new` into their command line, they will be asked these questions (defaults indicated by a `>`):
+
+- `Is this an app or an addon? (use arrow keys)`
+  - `> app`
+  - `addon`
+- `Please provide the name of your app/addon:`
+- `Please provide the spoken/content language of your app/addon: (use arrow keys)`
+  - `> en-US`
+  - `my computer's default language`
+  - `manually define a different language`
+- `Pick the package manager to use when installing dependencies:  (use arrow keys)`
+  - `> NPM`
+  - `Yarn`
+  - `ignore/skip`
+- `What CI system do you want to use? (use arrow keys)`
+  - `> travis-ci`
+  - `ignore/skip`
+
+#### App/Addon Name
+There will be no default for this question. The user must enter a response. 
+
+#### App/Addon Language
+
+- the default would be `en-US`
+- if your system has a different language set (which can be confirmed by using the `echo $LANG` command in the terminal), then _that_ language would be shown as the second option
+- if your system was already set to the default language, the second option would not be shown. 
+- for "manually define a language", we will validate the response against the allowed language codes
+- we have not added a "skip" option here, because the purpose is to focus on improved accessibility in Ember apps.
+
+#### Package Manager
+
+- ignore/skip has been added to cover the use case where the developer is in a workspace 
+
+#### CI Option
+
+Separate RFCs should further define more options for CI. 
+  
+### Current Limitations
+Explicitly, the goal of this RFC is to only include `ember-cli` flags that already exist. Any other possibilities should be considered future work.
+
+This interactive workflow will also ignore any ~/.ember-cli settings that already exist. A separate RFC should be written that makes it possible for these to things to co-exist and respect the settings of the other.
+
+### Future possibilities
+An example of a future RFC that could be specific to addons, would be to add a `--supported-version` flag for addons and add that to the wizard: 
+
+- `What is the earliest version of Ember that you intend to support? (use arrow keys)`
+  - `> recommended (last two LTS)`
+  - `last LTS`
+  - `manually define a version number`
```

Comment:
From above:

> What is the maintenance cost? *For example, a nice question to add could be something like, Do you want to use a CSS preprocessor? But, because Ember does not deal with SASS, LESS... by default, the interactive workflow will have to install an addon (like `ember-cli-less`) which would imply that the CLI team will have to ensure it never fails.* 

I can say, this is absolutely a nice idea but hardly manageable. A future possibility is to provide hooks for enhancing this by installing more addons (globally or per project). Vue cli did this with v2 already, was happy to see it when I was reading their code.

Reply:
What's the part that you think is not manageable? can you clarify what makes it not manageable? Thanks!

Reply:
Hard to manage a selection of possible options for addons to install. It's always subjective to the author(s) (if lectored) but I don't think will provide individuals/comapnies a good option. That's what I think, working on an API so you can _customize_ it for yourself or your companies is hitting the mark - although I see this as a future addition, not even tomorrow but farther future

Reply:
It's impossible to accommodate to everyone's needs, but I'd say it's good idea to provide with an easily accessible set of commonly agreed useful options. I know that I want addons XY, because I'm comfortable with using them. In my company we know that we want AB, because that's what we've chosen.

But person that is new to the framework or even JS ecosystem in general will have no idea where to start. At that point being given hints that "you might want sass if you want flexible styles", "you might want eslint if you want linting", "you might want prettier if you want unified code formatter", etc. is a great help to get started.

In the future it might be a good idea to abstract this wizard functionality and let the users use external sets. But I would not over-engineer this at this point.

Reply:
@gossi Ahh yes, you were agreeing with the stated position then. We decided that it would be not manageable, which is why we didn't include it. 👍

Reply:
Yeah... Sometimes I speak in riddles. I'm pretty sure this question will arise sooner than later. I think ember shouldn't provide any answers here at all (as this would be prejudgement). Instead what I'm saying is - to reach there - it will be nice if ember would provide an API to hook into. That way people can install there own flavors (eg. `yarn install -g ember-cli-new-config-css-modules`) and then it will be picked up in the wizard, to give an idea. Truly the (over)next step.